napari/layers/transforms.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations  # noqa: F407
+
+import toolz as tz
+from typing import Sequence
+import numpy as np
+
+from ..utils.list import ListModel
+
+
+class Transform:
+    """Base transform class.
+
+    Defaults to the identity transform.
+
+    Parameters
+    ----------
+    func : callable, Coords -> Coords
+        A function converting an NxD array of coordinates to NxD'.
+    """
+
+    def __init__(self, func=tz.identity, inverse=None):
+        self.func = func
+        self._inverse_func = inverse
+        if func is tz.identity:
+            self._inverse_func = tz.identity
+
+    def __call__(self, coords):
+        """Transform input coordinates to output."""
+        return self.func(coords)
+
+    def set_slice(self, axes: Sequence[int]) -> Transform:
+        """Return a transform subset to the visible dimensions."""
+        raise NotImplementedError('Cannot subset arbitrary transforms.')
+
+    @property
+    def inverse(self):
+        if self._inverse_func is not None:
+            return Transform(self._inverse_func, self.func)
+        else:
+            raise ValueError('Inverse function was not provided.')
+
+
+class TransformChain(Transform, ListModel):
+    def __init__(self, transforms=[]):
+        super().__init__(basetype=Transform, iterable=transforms, lookup=id)
```

Comment:
i dont think you want to specify a lookup table in this case

it was mainly meant for the layer list so it can be indexed by the layer names

Reply:
Thanks for the tip! I didn't know whether I needed one regardless. I think eventually indexing by name might be very useful, e.g. `grid` for all the transforms related to grid mode, so that they can be popped in and out as needed.